src/views/forms/TradeForm/TradePercentSizeToggle.tsx

```diff
@@ -0,0 +1,60 @@
+import styled from 'styled-components';
+
+import { TradeInputField } from '@/constants/abacus';
+import { ButtonShape } from '@/constants/buttons';
+import { PERCENT_DECIMALS } from '@/constants/numbers';
+
+import { formMixins } from '@/styles/formMixins';
+
+import { ToggleGroup } from '@/components/ToggleGroup';
+
+import abacusStateManager from '@/lib/abacus';
+import { MustBigNumber } from '@/lib/numbers';
+import { objectEntries } from '@/lib/objectHelpers';
+
+const PERCENTAGE_OPTIONS = {
+  '10%': 0.1,
+  '25%': 0.25,
+  '50%': 0.5,
+  '75%': 0.75,
+  '100%': 1,
+};
+
+type ElementProps = {
+  balancePercentValue: string;
+  setBalancePercentInputValue: (value: string) => void;
+};
+
+export const TradePercentSizeToggle = ({
+  balancePercentValue,
+  setBalancePercentInputValue,
+}: ElementProps) => {
+  const updateBalancePercent = (balancePercent: string) => {
+    setBalancePercentInputValue(balancePercent);
+
+    abacusStateManager.setTradeValue({
```

Comment:
hmm why doesn't the parent do this? This component api feels like it is a pure component but it has this kinda aggressive side effect.

Reply:
ah so I was mainly mimicking the [pattern](https://github.com/dydxprotocol/v4-web/blob/main/src/views/forms/TradeForm/TradeSizeInputs.tsx#L99) for the other inputs (usdcSize, size, etc) for consistency, but in this case the parent could update it itself. I think the reason we have this pattern for the other size inputs is due to formatting issues (you want to set the value to be the floatValue rather than the formattedValue - but that's what you want to display). Technically this isn't an issue for the balance percent toggles (there's nothing to format). 

I could move it to something similar to [`useTradeFormInputs`](https://github.com/dydxprotocol/v4-web/blob/main/src/hooks/useTradeFormInputs.ts) (but we don't have a `useTradeSizeInputs` for the reasons above, so I think for consistency it's maybe easiest to reason about if we keep it here 🤔 what do ya think tho?

Reply:
I suppose it's okay, but the reason this one is weird and the other one isn't is that this has the component api of a controlled component but has a specific side effect in here. TradeFormInput doesn't take any props so it's clear it's handling its own business.

Reply:
Yeah I think it's weird to do this here.

Reply:
hmm ok lemme [follow up](https://linear.app/dydx/issue/CT-1257/[tech-debt]-move-out-setting-of-balancepercent-into-parent) on this!